### PR TITLE
Send a Shoutout

### LINKED ISSUE #91: Consider using Twitch's shoutout
Sends a Shoutout to the specified broadcaster. Typically, you send Shoutouts when you or one of your moderators notice another broadcaster in your chat, the other broadcaster is coming up in conversation, or after they raid your broadcast: https://dev.twitch.tv/docs/api/reference/#send-a-shoutout

image
image
image
image

**Authorization**
Requires a [user access token](https://dev.twitch.tv/docs/authentication#user-access-tokens) that includes the `moderator:manage:shoutouts` scope. This scope needs to be added to:

https://github.com/whitep4nth3r/p4nth3rb0t-mainframe/blob/f5e31d56ee36d9b99d581a871a2520de2046104e/src/classes/AccessToken.ts#L42

**URL**
POST `https://api.twitch.tv/helix/chat/shoutouts`

**Request Query Parameters**
Parameter | Type | Required? | Description
-- | -- | -- | --
from_broadcaster_id | String | Yes | The ID of the broadcaster that’s sending the Shoutout.
to_broadcaster_id | String | Yes | The ID of the broadcaster that’s receiving the Shoutout.
moderator_id | String | Yes | The ID of the broadcaster or a user that is one of the broadcaster’s moderators. This ID must match the user ID in the access token.

### LINKED ISSUE #96: Announce with p4nth3rb0t instead of broadcaster
Currently we are announcing using the broadcaster account.

It would be preferable to announce using the `config.bot.user_id`, but this requires some access token configuration.

### PR DESCRIPTION
**Pull request by ricardodevries**
This pull request closes #91 and resolves #96.

### CODE REVIEW

**Comment by whitep4nth3r** [comment]
I am having trouble accepting that this PR changes the way we do authentication from the broadcaster account to the bot account, mainly because it's a big hassle for me to log into the bot account and authenticate quickly and easily.

**Comment by ricardodevries** [comment]
> I am having trouble accepting that this PR changes the way we do authentication from the broadcaster account to the bot account, mainly because it's a big hassle for me to log into the bot account and authenticate quickly and easily.

That is why I posted about both ways. If you want to keep authenticating with your broadcaster account then you could apply the suggestion I made in this PR.

**Comment by whitep4nth3r** [comment]
I'll test this locally when I get chance. Thank you!

**Comment by ricardodevries** [comment]
Ready for merge @whitep4nth3r

**Review by jwalter — APPROVED** [review]

File: src/utils/twitchUtils.ts [hunk 1]
```diff
@@ -130,6 +130,15 @@ export const announce = async (
   }
 };
 
+export const shoutOut = async (toBroadcasterId: string): Promise<any> => {
+  const url = `https://api.twitch.tv/helix/chat/shoutouts?from_broadcaster_id=${config.broadcaster.user_id}&to_broadcaster_id=${toBroadcasterId}&moderator_id=${config.bot.user_id}`;
+  const response = await doPost(url, null);
+
+  if (response) {
+    console.log(`Shoutout to ${toBroadcasterId}`);
+  }
```
Comment (whitep4nth3r) [inline src/utils/twitchUtils.ts:144]: Shall we add some error tracking here, and send to Sentry in a try catch?

**Pull request merged**